### PR TITLE
Add support for containerized badfish and parallel execution

### PR DESCRIPTION
**Pull request by rajeshP524**
Made changes for using containerized badfish and for parallel execution
when we have list of hosts for setting boot order, badfish can take list
of all hosts at once and operate on each simultaneously.

### CODE REVIEW

File: badfish/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,2 @@
+FROM quay.io/quads/badfish
+COPY bad-hosts /
```
Comment (sanjaychari) [inline badfish/Dockerfile]: Please change the name from `bad-hosts` to `dell-hosts` or something similar. bad-hosts is a misleading name :).

File: common.yml [hunk 2]
```diff
@@ -65,6 +65,27 @@
...
+            state: absent
+
+      - name: prepare dell hosts for badfish
+        shell: echo "mgmt-{{ item[1].stdout }}" >> ~/jetpack/badfish/bad-hosts
+        when: vendors is defined and item[0] == "dell"
+        with_together:
+          - "{{ vendors }}"
+          - "{{ host_list.results }}"
+
+      - name: remove existing badfish image
```
Comment (sanjaychari) [inline common.yml:80]: If the image is up to date, I don't think we need to remove it. It will simply lead to additional time being spent. Maybe you can change this step to remove the existing image only if the image is not the latest version.

**Review by sanjaychari — CHANGES_REQUESTED** [review]

File: badfish/Dockerfile (hunk 1 above)
Comment (rajeshP524) [reply badfish/Dockerfile]: I took this term from badfish repository, in which they have used it for an example. using the name 'bad-hosts' with badfish commands would be meaningful.

File: common.yml (hunk 2 above)
Comment (rajeshP524) [reply common.yml:80]: The existing image will contain list of dell hosts from the previous deployment. so, pulling it freshly will add the new hosts.
refer to jetpack/badfish/Dockerfile from this PR

Comment (sanjaychari) [reply common.yml:80]: Ack

File: badfish/Dockerfile (hunk 1 above)
Comment (sanjaychari) [reply badfish/Dockerfile]: `bad-hosts` sounds like it's referring to hosts which are faulty/bad. `dell-hosts` or something else like `badfish-hosts` would be more appropriate.

Comment (rajeshP524) [reply badfish/Dockerfile]: okay. will update it to 'dell-hosts'.

File: common.yml [hunk 3]
```diff
@@ -65,6 +65,27 @@
         with_sequence: 0-{{ (oc_instackenv_content.nodes|length - 1) }}
         register: host_list
 
+      - name: remove existing bad-hosts file
+        file:
+            path: "~/jetpack/badfish/bad-hosts"
```
Comment (asyedham) [inline common.yml]: please use a appropriate naming something like dell-hosts as bad-hosts would indicate faulty nodes

File: tasks/set_boot_order_director.yml [hunk 4]
```diff
@@ -1,70 +1,35 @@
...
-    source {{ badfish_venv }}/bin/activate;
-    python3 badfish.py -H mgmt-{{ item[1].stdout }}  -u quads -p {{ chassis_password }} --clear-jobs --force
-  args:
-    chdir: "{{ ansible_user_dir }}/badfish/src/badfish"
-  when: vendors is defined and item[0] == "dell"
-  with_together:
-    - "{{ vendors }}"
-    - "{{ host_list.results }}"
+  become: yes
+  shell: podman run -it --rm quay.io/quads/badfish --host-list /bad-hosts -u quads -p {{ chassis_password }} --clear-jobs --force
```
Comment (asyedham) [inline tasks/set_boot_order_director.yml]: It would be better if you create separate PR's for badfish container and parallel execution to set boot order for readability

**Review by asyedham — CHANGES_REQUESTED** [review]

File: common.yml (hunk 3 above)
Comment (rajeshP524) [reply common.yml]: sure

**Comment by sanjaychari** [comment]
Mention that this PR resolves https://github.com/redhat-performance/jetpack/issues/452.

**Comment by rajeshP524** [comment]
This PR has been split into separte PR's https://github.com/redhat-performance/jetpack/pull/457 and https://github.com/redhat-performance/jetpack/pull/459
